src/Compatibility/Microsoft.DotNet.PackageValidation/ApiCompatRunner.cs

```diff
@@ -87,6 +99,58 @@ public void RunApiCompat()
             _dict.Clear();
         }
 
+        private IAssemblySymbol GetAssemblySymbolFromStream(Stream assemblyStream, MetadataInformation assemblyInformation, out bool resolvedReferences)
+        {
+            resolvedReferences = false;
+            HashSet<string> directories = null;
+
+            // In order to enable reference support for baseline suppression we need a better way
+            // to resolve references for the baseline package. Let's not enable it for now.
+            bool shouldResolveReferences = !_isBaselineSuppression && _referenceDirectories != null &&
+                _referenceDirectories.TryGetValue(assemblyInformation.TargetFramework, out directories);
+
+            AssemblySymbolLoader loader = new(resolveAssemblyReferences: shouldResolveReferences);
+            if (shouldResolveReferences)
+            {
+                resolvedReferences = true;
+                loader.AddReferenceSearchDirectories(directories);
+            }
+            else if (!_isBaselineSuppression && _referenceDirectories != null)
+            {
+                _log.LogWarning(
```

Comment:
So far we have used the warnings mostly for compatibility validation errors. This case is a bit different as this is mainly just a message that says that we are not using Warnings for this specific check. Should this be informational message as opposed to warning? @ericstj thoughts?

Reply:
If the user has communicated to us that they wish to use dependencies, but has not provided those dependencies then we need to tell the user about this.  I don't think a message is strong enough.  Error's are not suppressible.  I do think it makes sense to use a suppressible warning.